logging/bin/change_internal_password.sh

```diff
@@ -61,56 +65,34 @@ if [ "$NEW_PASSWD" == "" ]; then
    autogenerated_password="true"
 fi
 
-#get current credentials from Kubernetes secret
-if [ -z "$(kubectl -n $LOG_NS get secret internal-user-$USER_NAME -o=name 2>/dev/null)" ]; then
-   log_warn "The Kubernetes secret [internal-user-$USER_NAME], containing credentials for the user, was not found."
-   # TO DO: How to handle case where secret does not exist?  Should never happen. 
-   # exit
-   ES_USER=$USER_NAME
+if [ "$USER_NAME" != "logadm" ]; then
+   #get current credentials from Kubernetes secret
+   if [ -z "$(kubectl -n $LOG_NS get secret internal-user-$USER_NAME -o=name 2>/dev/null)" ]; then
+      log_warn "The Kubernetes secret [internal-user-$USER_NAME], containing credentials for the user, was not found."
+      # TO DO: How to handle case where secret does not exist?  Should never happen. 
+      # exit
+      ES_USER=$USER_NAME
+   else
+      ES_USER=$(kubectl -n $LOG_NS get secret internal-user-$USER_NAME -o=jsonpath="{.data.\username}" |base64 --decode)
+      ES_PASSWD=$(kubectl -n $LOG_NS get secret internal-user-$USER_NAME -o=jsonpath="{.data.password}" |base64 --decode)
+   fi
 else
-   ES_USER=$(kubectl -n $LOG_NS get secret internal-user-$USER_NAME -o=jsonpath="{.data.\username}" |base64 --decode)
-   ES_PASSWD=$(kubectl -n $LOG_NS get secret internal-user-$USER_NAME -o=jsonpath="{.data.password}" |base64 --decode)
+  ES_USER=$USER_NAME
+  ES_PASSWD="do_not_know_current_password"
```

Comment:
I assume this is a placeholder. What happens once this is set and why use this value? I don't see another reference to the value.

Reply:
ES_PASSWD is the current password for the account and used in the curl command that follows.  That curl uses the user's own credentials to change the password. But, since we don't know the current password for the logadm account, I was providing a dummy value so the curl command wouldn't fail due to a syntax error.  We want the processing to fall through to the use the admin user's credentials to change the logadm user password.  

Using a nonsense value here was the easiest implementation and avoided having to restructure the code more extensively to handle the logadm account.